### ISSUE TITLE
Annotations for CRT init function pointers

### ISSUE THREAD

**Issue by disinvite** [isledecomp/reccmp#37]
Back in November, @madebr asked:
> Does reccmp have a way to verify global initializations/destructions?
__attribute((constructor))/__attribute((destructor)) for gcc, and function pointers in the .CRT$XC(V|T) section?
https://learn.microsoft.com/en-us/cpp/c-runtime-library/crt-initialization
> This is usually done by _initterm.

And:
> Let's consider this global:
> ```cpp
> // GLOBAL: LEGO1 0x0123467
> std::string g_actor_name = "brickster";
> ```
> The global only tells us where g_actor_name lives, but we want to also check whether the (non-c++) constructor call is identical.
> So I think it also needs a // CONSTRUCTOR: LEGO1 ... annotation.
> So is this a combination of decomplint and reccmp?
> Do PDB's have information about this?

In `LEGO1`, there are 7 of these pointers between `xc_a` and `xc_z` in `.data`. The first sets an exit callback with `atexit` and calls `KERNEL32.DLL::SetUnhandledExceptionFilter`. No name in the PDB.

The other 6 are thunks that point to the constructor functions. All 12 of those appear to correspond to the `S_LPROC32` symbols in cvdump output.

The 6 constructor functions have a corresponding line reference in the `LINES` section, so it would be possible to add a new `CONSTRUCTOR` marker type that could run alongside variables and match these up.

**Comment by jonschz** [comment]
We have already collected a bunch of those references, but they aren't checked - see e.g. https://github.com/isledecomp/isle/blob/2d74f14de09b94f41d2d6bbb47537163b35146a1/LEGO1/lego/legoomni/src/race/legoracers.cpp#L170

**Comment by disinvite** [comment]
I don't think this will be difficult to add, but we probably want a name other than `// CONSTRUCTOR`. Not that this is wrong, but I could see someone marking constructor functions with that comment by mistake.

How about just `// CRT_INIT` ?

**Comment by disinvite** [comment]
I'm working on a way to identify and match these functions without any new annotations. The only requirement is that the `___xc_a` and `___xc_z` labels are matched. We know that each pointer between those values is a function. For each of those we:
- Disassemble and collect each pointer/address used in the function.
- Of those, if the value is in our database _and matched_, add the address from the original binary to a list.
- This collection of addresses is our "fingerprint" for the dynamic initializer function.

Then match functions that have the same fingerprint in orig and recomp.
